### DISCUSSION TITLE
Snow Leopard?

### DISCUSSION THREAD

**Discussion by cmhjon** [utmapp/UTM#6500] [macOS]
Hello all. I am new to the forum. First, let me say that I am VERY impressed with UTM! Running Windows 11 and an older XP VM without issue!

That said, I would like to add a macOS 10.6 "Snow Leopard" UTM but am not sure which settings I should use. Can someone help me with that?

I did review the content in (https://github.com/utmapp/UTM/discussions/2465#discussioncomment-9427909) and https://github.com/royalgraphx/LegacyOSXKVM/blob/main/info/AS_SL.md but I'm afraid I don't understand what is mentioned in the second link.

Thank you in advance,
Jon :-)

**Comment by xadammr** [comment]
I have been trying to get Snow Leopard working in UTM for quite a while now.

I had a look at the links you provided. I downloaded the pre-prepared images from the LegacyOSXKVM repo, and I still get a panic on boot. 

![image](https://github.com/user-attachments/assets/e9af0b8a-90c4-4324-86eb-1387d2b613ae)

**Reply by cmhjon** [reply]
As it turns out, Leopard is not going to work for me as one app I need to install requires Snow Leopard. I see below, Daniel mentioned an install by IAtkos but don’t know where to go from here. I found a lot of references to IAtkos but not sure what or where to download, let alone the settings needed to setup the UTM.

**Comment by cmhjon** [comment]
Hi xadammr, I am giving up on Snow Leopard so I will settle for Leopard. Can you provide step by step instructions (including everything I need to download and all settings) on how to add a Leopard VM via UTM running on an M1 Mac?

**Comment by cmhjon** [comment]
I found and downloaded a full Leopard UTM from here:

https://archive.org/details/mac-os-ppc-utm-vms

**Comment by Denial-Steinberg** [comment]
IAtkos s2 snow leopard +- works on UTM m1.

**Reply by cmhjon** [reply]
Hi Daniel. Thank you for the reply! That's great news! Where can I find IAtkos' configuration and settings?

**Reply by Denial-Steinberg** [reply]
Hello cmhjon.  I can’t say whether there was one version of iAtkos_S3_v2 or whether there were many of them; I downloaded it 10 years ago.  I was unable to install, but I was able to login to the installation. Settings UTM 86_64, usb 3, vmware vga.
I didn't have desktop. Next you need to do experiments with kernels and kexts.

**Comment by xadammr** [comment]
@cmhjon I have worked out the magic incantations! I wrote up my findings at https://xadammr.au/blog/snow-leopard-utm-apple-silicon -- but long story short, Snow Leopard wants `-global ICH9-LPC.disable_s3=0` added to the QEMU arguments in UTM.

**Reply by cmhjon** [reply]
Hi Adam!

This is fantastic! Thank you so very much for your help! Using your instructions, I managed to get a bootable Snow Leopard UTM! Now I have some questions I am hoping you can help me with:

1. When I go to Preferences and enable the “Hard Disks” checkbox under the “Show these items on the desktop:” section, although the icons appear, I get a kernel panic. Any ideas on how to eliminate the panic? I really would like the icons to be visible if at all possible.
2. Now that the Snow Leopard UTM is booted and working, do I still need to hold onto the ‘AS_SL.img' and the ‘macintosh_hd.img' files that were downloaded and imported?
3. When exiting the System Preferences app, I naturally press CMD-Q but instead of the System Preferences app quitting, the UTM gets shutdown and the UTM app closes. Is there a way to pass the keyboard commands to Snow Leopard rather than the UTM app itself?

Again, thank you thank you thank you and best regards,
Jon :-)

**Reply by xadammr** [reply]
Hi Jon

1. Sorry, I can't reproduce this -- I am able to show and hide the hard disks on the desktop to my hearts content.
2. You can definitely delete what you had downloaded -- UTM converts them to `qcow2` format and saves it inside the virtual machine bundle.
3. That is a good question, and took me a second to work out. If you make sure the mouse is captured (press ⌃ + ⌥ to capture), all keyboard shortcuts will be sent to the VM rather than the app.

**Reply by cmhjon** [reply]
Hi Adam,

Rebooting the host Mac seems to have resolved the panic when enabling the aforementioned checkbox. Pressing ⌃ + ⌥ did indeed take care of the # 3. Now for two more questions:

1. I am using an M1 iMac. The only available resolution in the Displays System Preference is 1280 x 720. Is there a way to get more choices? I tried option-clicking the green button in the upper left corner of the window but all that did was keep the same res and make everything bigger (and not long after option-clicking the green button again, I got another kernel panic).
2. I assume I should NOT try to update Snow Leopard to 10.6.8?

I appreciate your suffering through all my questions!

Best regards,
Jon :-)

**Reply by xadammr** [reply]
Hi Jon

1. In your /Volumes/EFI/EFI/OC/config.plist, set `UEFI > Output > ForceResolution` to `TRUE`, and that should unlock all the resolution options in System Preferences. (Edit: the screenshot is from Xcode, but if you don't have it installed you can just use TextEdit and search for the string `ForceResolution` to locate the line in question.)

![image](https://github.com/user-attachments/assets/571fb5f9-ee0c-47c3-abe4-2d5ec364a6ac)
![image](https://github.com/user-attachments/assets/3c274421-43c3-4850-8e6b-87e94ccc9c62)

2. No problems on my end updating to 10.6.8. Probably worth confirming on your end by running without saving changes and doing the update, to confirm nothing you need breaks:

![image](https://github.com/user-attachments/assets/4dde8a11-07c5-4d38-9cbc-117d9dfb3691)

No problems with all the questions! 

-A

**Reply by cmhjon** [reply]
Hi Adam,

I made the change to the 'ForceResolution' key, saved, and rebooted. Still only have one resolution available. Maybe I didn't make the change correctly?

![Screenshot 2025-01-21 at 7 30 05 AM](https://github.com/user-attachments/assets/823eb346-49a8-4401-a9d0-93a25d660eb4)

Thank you and best regards,
Jon :-)

**Reply by cmhjon** [reply]
Hi Adam,

Quick update. I noticed that a few keys below "ForceResolution" was a "Resolution" key with a value of "1280x720". After making a backup of the UTM, I tried changing this value to "1920x1080" (which is the resolution of a Leopard UTM I have) and it seems to have worked! Although I would still like to know why I am not getting a list of all the resolutions like your UTM, if setting this key manually is safe(?), I am ok with it.

Best regards,
Jon :-)

**Reply by cmhjon** [reply]
Hi Adam,

Have you had any panics at all? Mine was working decently and then all of a sudden, I started getting recurring panics at different times. I hadn't done a lot with mine other than update it to 10.6.8 and get it connected to my internal network. Fortunately, I had a backup of the UTM before I did anything to it and it was panicking as well.

If you haven't, would you be willing to post your full Snow Leopard UTM somewhere that I could download it? I am going to try redownloading the IMG file from archive.org. Last time it took over 9 hours to download and I wonder if perhaps something got slightly corrupted during the download.

Best regards,
Jon :-)

**Reply by xadammr** [reply]
Hey Jon -- sorry I missed this new comment. I see you commented via my blog post as well. If you can contact me via an alternate means, I'll be able to share a link to my Snow Leopard VM.

**Reply by cmhjon** [reply]
Hi Adam! Thank you for the reply. Although I welcome your VM, I am not sure how to contact you via alternate means. I’ve tried searching all your social media pages as well as GitHub but am unable to find an email address anywhere. I did find your profile on LinkedIn and will send you a message there.

**Reply by Denial-Steinberg** [reply]
Can you please send it to me too, I tried to configure it but nothing worked.

**Reply by MickBehn** [reply]
Hi Jon and Adam,
I can't get the resolutions to show up in Snow leopard. it's stuck at 1280. so I tried Adams solution of 1920x1080 the Resolution field. but its away to unstable, kernel panics and super slow on an M3 Max. I figured that might be because 1920x1080 isn't a native OSX resolution, so I changed it to 1440x900 which is the MacBook Pro 2012 15" native resolution and it works great super fast and stable.  I think Snow Leopard is just sensitive to native resolutions. that said I'm running the Snow leopard server 10.6.8 version, so it might be different than the standard Snow leopard. thought that might help.

**Reply by startergo** [reply]
> Hi Jon and Adam, I can't get the resolutions to show up in Snow leopard. it's stuck at 1280. so I tried Adams solution of 1920x1080 the Resolution field. but its away to unstable, kernel panics and super slow on an M3 Max. I figured that might be because 1920x1080 isn't a native OSX resolution, so I changed it to 1440x900 which is the MacBook Pro 2012 15" native resolution and it works great super fast and stable. I think Snow Leopard is just sensitive to native resolutions. that said I'm running the Snow leopard server 10.6.8 version, so it might be different than the standard Snow leopard. thought that might help.

[1920x1080-OpenCoreSLeopard.qcow2.zip](https://github.com/user-attachments/files/20824093/1920x1080-OpenCoreSLeopard.qcow2.zip)
Try this EFI folder:
![image](https://github.com/user-attachments/assets/7a16cd93-874f-4b12-a881-3763374992ed)

**Comment by Denial-Steinberg** [comment]
Friends. I do everything according to the instructions one by one but I get a kernel panic. Can you please upload the UTM file. and a file with arguments. I will be very grateful.

**Reply by startergo** [reply]
[config.plist.zip](https://github.com/user-attachments/files/20824590/config.plist.zip)
[command.txt](https://github.com/user-attachments/files/20824606/command.txt)

**Comment by barracuda156** [comment]
@xadammr Could you say if Rosetta works? I need powerpc support in 10.6.8, but probably it is easier to do that via Rosetta rather than emulate G4 and try to get native 10.6.8 running.
